source/MaterialXCore/Types.h

```diff
@@ -173,6 +174,28 @@ template <class V, class S, size_t N> class VectorN : public VectorBase
         return *this;
     }
 
+    /// @}
+    /// @name Geometric operators
+    /// @{
+
+    /// Magnitude of a vector
+    S magnitude() const
```

Comment:
For consistency and clarity, I would recommend naming these methods getMagnitude and getNormalized, where the latter method returns a new vector instead of modifying the vector in place.  There's a long history of functions named "normalize" having inconsistent behaviors, with some returning new vectors, and others normalizing the vector in place and returning the original magnitude.  For this library, I think we're better off avoiding this area of confusion entirely, and using a method name without any ambiguity.

Reply:
Will do.